lab-anthony/lib/read.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const fs = require('fs');
+
+module.exports = function(pathsArray) {
+  var dataArray = [];
+  pathsArray.forEach(function(file, index){
+    fs.readFile(file, (err, data) => {
+      if (err) throw err;
+      dataArray[index] = data.toString('hex', 0, 8);
+
+      if (pathsArray.length === dataArray.length)
```

Comment:
This if statement isn't doing what you're thinking it does--it does not ensure that this is only done once all of the files have finished. Try this in the console:
```javascript
var test = [];
test[3] = 'foo';
console.log(test.length);
```